Pull Request to merge with myexpr/raml-diff

harishputhran opened the pull request.

harishputhran committed:
Added Code changes to compare Trait information in RAML files
Added Parent URI to Resource Relative Path during creation of ActionId.

harishputhran committed:
Updated the existing JUnits to resolve the issues.
Added method to mock Action for setting Traits informations in Action
objects.

harishputhran committed:
1. Modified the Code to remove the unnecessary null checks.
2. Created the Test Class for FindActionsWithUpdatedTraits.
3. Created the Test Class for RamlDiffEngineImpl With scenarios for Only
Trait Finder changes.
4. Added methods to MockHelper to create a Collection<Resource> from
Raml File for test scenarios
5. Added Uri parameter in place of RelativeUri during creation of
ActionId objects
6. Applied the GoogleStyle Eclipse Code Formatting

myexpr commented:
Will do this later in the day.

Thanks
On May 13, 2015 10:05, "Harish" <notifications@github.com> wrote:

> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/myexpr/raml-diff/pull/2
> Commit Summary
>
>    - Added Code changes to compare Trait information in RAML files
>    - Updated the existing JUnits to resolve the issues.
>    - 1. Modified the Code to remove the unnecessary null checks.
>
> File Changes
>
>    - *A* .settings/org.eclipse.jdt.ui.prefs
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-0> (3)
>    - *A* pom.xml <https://github.com/myexpr/raml-diff/pull/2/files#diff-1>
>    (67)
>    - *A* src/main/java/diff/TraitDiff.java
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-2> (21)
>    - *M* src/main/java/engine/impl/RamlDiffEngineImpl.java
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-3> (80)
>    - *M*
>    src/main/java/engine/impl/finders/AbstractFindActionsWithUpdatedParameters.java
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-4> (122)
>    - *A*
>    src/main/java/engine/impl/finders/FindActionsWithUpdatedTraits.java
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-5> (69)
>    - *M* src/main/java/service/ActionIdService.java
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-6> (38)
>    - *M* src/main/java/service/RamlDiffService.java
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-7> (161)
>    - *M* src/test/java/engine/MockHelper.java
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-8> (193)
>    - *M* src/test/java/engine/impl/RamlDiffEngineImplTest.java
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-9> (138)
>    - *A*
>    src/test/java/engine/impl/RamlDiffEngineImplWithOnlyTraitFinderTest.java
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-10> (98)
>    - *A*
>    src/test/java/engine/impl/finders/FindActionsWithUpdatedTraitsTest.java
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-11> (83)
>    - *A* src/test/resources/complete_raml.raml
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-12> (21650)
>    - *A*
>    src/test/resources/complete_raml_all_fiterable_traits_removed.raml
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-13> (21650)
>    - *A*
>    src/test/resources/complete_raml_all_historical_trait_removed.raml
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-14> (21650)
>    - *A*
>    src/test/resources/complete_raml_one_historical_and_filter_traits_removed.raml
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-15> (21650)
>    - *A*
>    src/test/resources/complete_raml_with_one_resource_with_trait.raml
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-16> (21650)
>    - *A*
>    src/test/resources/complete_raml_with_one_resource_with_trait_one_parameter_added_to_trait.raml
>    <https://github.com/myexpr/raml-diff/pull/2/files#diff-17> (21655)
>
> Patch Links:
>
>    - https://github.com/myexpr/raml-diff/pull/2.patch
>    - https://github.com/myexpr/raml-diff/pull/2.diff
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/myexpr/raml-diff/pull/2>.
>

harishputhran committed:
Added Code for Comparing Response details between RAML Files.
Added methods in Mock Helper for Junits related to Response comparison.
Added JUnit Test cases.
